### ISSUE TITLE
[Flutter Web] Splash Screen created using HTML and CSS overlays even after loading on released build.

### ISSUE THREAD

**Issue by Maverick099** [flutter/flutter#85914]
A simple Animated Splash screen made using CSS works as expected in Debug mode (disappears after the main website is loaded) but the released-mode website deployed on  https://coders-asylum.github.io/#/ shows the splash screen overlayed on the main website.  

release mode website created using this command in CI: `flutter build web --pub --release --null-assertions`
## Steps to Reproduce

1)  Add this to the index.html
```
<body>
.
.
.
<div class="center">
    <div class="title">
        <span class="dollar">$</span><span class="typewriter">loading...</span>
    </div>
</div>
</body>
```
refer this: https://github.com/Coders-Asylum/coders-asylum.github.io/blob/aae6aee405f816afc0c3b6ea822598a1292c175c/web/index.html#L46

2) Add this css :
```
.center {
    position: absolute;
    top: 50%;
    left: 50%;
    margin: 0 -50% 0 0;
    transform: translate(-50%, -50%)
}

html, body {
    background-color: #1E1E1E;
}

.image {
    position: relative;
}

.title {
    position: relative;
    font-family: "Gobold", sans-serif;
    font-size: 30px;
    color: white;
    padding-left: 20px;
    padding-top: 5px;
}

@font-face {
    font-family: SourceCodePro;
    src: url("../../lib/assets/fonts/SourceCodePro-Regular.ttf");
}

.dollar {
    font-family: "SourceCodePro", sans-serif;
    font-size: 30px;
    color: yellow;
    padding-left: 20px;
    padding-top: 5px;
}

.typewriter {
    overflow: hidden;
    border-right: .15em solid orange;
    white-space: nowrap;
    margin: 0 auto;
    letter-spacing: .15em;
    animation: typing 3.5s steps(40, end),
    blink-caret .75s step-end infinite;
}

@keyframes typing {
    from {
        width: 0
    }
    to {
        width: 100%
    }
}

@keyframes blink-caret {
    from, to {
        border-color: transparent
    }
    50% {
        border-color: orange;
    }
} 

```

refer this: https://github.com/Coders-Asylum/coders-asylum.github.io/blob/master/web/css/splash.css

Web application source code: https://github.com/Coders-Asylum/coders-asylum.github.io

**Expected results:** 
Splash Screen disappearing after loading:

![2021-07-05 14-46-43](https://user-images.githubusercontent.com/32545664/124464516-e8d2a180-ddb1-11eb-8c8d-454de1f882ba.gif)

**Actual results:** 

You can see the released website here: https://coders-asylum.github.io/#/

<table>
<tr>
<th align="center">
<img width="441" height="1">
<p> 
<small>
Debug mode output (working as expected)
</small>
</p>
</th>
<th align="center">
<img width="441" height="1">
<p> 
<small>
Released Website output
</small>
</p>
</th>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/32545664/124464875-55e63700-ddb2-11eb-8ac2-9c278824f6d8.gif" width="480" height="270" >
</td>
<td>
<img src="https://user-images.githubusercontent.com/32545664/124465218-c8571700-ddb2-11eb-8927-8c3b8a21b5af.gif" width="480" height="270">
</td>
</tr>
</table>
<details>
  <summary>Logs</summary>

```
[  +99 ms] executing: [C:\code\flutter\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +128 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +2 ms] d79295af24c3ed621c33713ecda14ad196fd9c31
[   +1 ms] executing: [C:\code\flutter\flutter/] git tag --points-at d79295af24c3ed621c33713ecda14ad196fd9c31
[ +173 ms] Exit code 0 from: git tag --points-at d79295af24c3ed621c33713ecda14ad196fd9c31
[        ] 2.2.2
[   +6 ms] executing: [C:\code\flutter\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +73 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\code\flutter\flutter/] git ls-remote --get-url origin
[  +73 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +638 ms] executing: [C:\code\flutter\flutter/] git rev-parse --abbrev-ref HEAD
[  +78 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +80 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +146 ms] executing: C:\Users\samar\AppData\Local\Android\Sdk\platform-tools\adb.exe devices -l
[ +106 ms] List of devices attached
[  +14 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +128 ms] Multiple devices found:
[  +51 ms] Chrome (web) • chrome • web-javascript • Google Chrome 91.0.4472.124
[   +4 ms] Edge (web)   • edge   • web-javascript • Microsoft Edge 91.0.864.59
[   +4 ms] [1]: Chrome (chrome)
[   +1 ms] [2]: Edge (edge)
[   +4 ms] Please choose one (To quit, press "q/Q")
[   +1 ms] : 
[+9235 ms] 1
[  +29 ms] Skipping pub get: version match.
[ +362 ms] Launching lib\main.dart on Chrome in debug mode...
[ +280 ms] Updating assets
[ +213 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
[   +3 ms] Waiting for connection from debug service on Chrome...
[ +719 ms] <- reset
[   +8 ms] C:\code\flutter\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev
C:\code\flutter\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\code\flutter\flutter\bin\cache\flutter_web_sdk/
--incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
C:\Users\samar\AppData\Local\Temp\flutter_tools.57ce7201\flutter_tool.5ae9b75\app.dill --libraries-spec
file:///C:/code/flutter/flutter/bin/cache/flutter_web_sdk/libraries.json --packages D:\work_src\Medium Coders Asylum\website\.dart_tool\package_config.json
-Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root
C:\Users\samar\AppData\Local\Temp\flutter_tools.57ce7201\flutter_tools.c231896c --filesystem-scheme org-dartlang-app --initialize-from-dill
build\b1b715402d823b7fd5c2b68d2edcb2ce.cache.dill.track.dill --platform file:///C:/code/flutter/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk_sound.dill
--sound-null-safety
[  +17 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+24944 ms] Waiting for connection from debug service on Chrome... (completed in 25.7s)
[   +1 ms] Synced 28.7MB.
[   +1 ms] <- accept
[   +1 ms] Caching compiled dill
[ +879 ms] [CHROME]:
[  +20 ms] [CHROME]:DevTools listening on ws://127.0.0.1:59164/devtools/browser/e827dff8-23eb-40c7-8258-7b285d9ee584
[ +871 ms] DwdsInjector: Received request for entrypoint at http://localhost:60126/main_module.bootstrap.js
[  +11 ms] MetadataProvider: Loading debug metadata...
[  +22 ms] MetadataProvider: Loaded debug metadata (sound null safety)
[   +9 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:60126/main_module.bootstrap.js
[+2785 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: true
[  +99 ms] DevHandler: Debug service listening on ws://127.0.0.1:64682/fFjGtr77v6g=/ws

[  +32 ms] This app is linked to the debug service: ws://127.0.0.1:64682/fFjGtr77v6g=/ws
[  +13 ms] Debug service listening on ws://127.0.0.1:64682/fFjGtr77v6g=/ws
[        ]  Running with sound null safety 
[   +3 ms]   To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
[+9881 ms] Performing hot restart...
[   +5 ms] Initializing file store
[  +14 ms] Skipping target: gen_localizations
[   +5 ms] complete
[ +130 ms] Scanned through 528 files in 127ms
[   +1 ms] Waiting for connection from debug service on Chrome...
[   +2 ms] <- recompile org-dartlang-app:/web_entrypoint.dart ff9e4373-75f0-4763-aaf8-3ace636e8126
[        ] <- ff9e4373-75f0-4763-aaf8-3ace636e8126
[  +19 ms] Waiting for connection from debug service on Chrome... (completed in 21ms)
[        ] Synced 0.0MB.
[        ] <- accept
[   +7 ms] DwdsVmClient: Attempting a hot restart
[   +1 ms] DwdsVmClient: Attempting to disabling breakpoints and resume the isolate
[   +8 ms] DwdsVmClient: Successfully disabled breakpoints and resumed the isolate
[   +1 ms] DwdsVmClient: Attempting to get execution context ID.
[        ] DwdsVmClient: Got execution context ID.
[        ] DwdsVmClient: Issuing $dartHotRestart request.
[  +63 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: true
[  +19 ms] DwdsVmClient: $dartHotRestart request complete.
[        ] DwdsVmClient: Waiting for Isolate Start event.
[   +2 ms] DwdsVmClient: Successful hot restart
[   +2 ms] Performing hot restart... (completed in 286ms)
[        ] Restarted application in 289ms.


```

```
Analyzing website...
No issues found! (ran in 3.6s)

```

```
[√] Flutter (Channel stable, 2.2.2, on Microsoft Windows [Version 10.0.19043.1081], locale en-IN)
    • Flutter version 2.2.2 at C:\code\flutter\flutter
    • Framework revision d79295af24 (3 weeks ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\samar\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[√] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.3
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.57.1)
    • VS Code at C:\Users\samar\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.24.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 91.0.4472.124
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 91.0.864.59

• No issues found!


```

</details>

**Comment by TahaTesser** [comment]
Hi @Maverick099 
Just tried to reproduce this issue in a new project on the latest `stable` channel. "Splash screen" doesn't overlay 


https://user-images.githubusercontent.com/48603081/124596864-6e665800-de6b-11eb-8532-d35f91ce2173.mp4


### Project files
[stable_flutter.zip](https://github.com/flutter/flutter/files/6769852/stable_flutter.zip)


<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.22000.51], locale en-US)
    • Flutter version 2.2.3 at C:\Users\Taha\Code\flutter_stable
    • Framework revision f4abaa0735 (5 days ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java.exe
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31424.327
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] VS Code (version 1.57.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.24.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.51]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 91.0.4472.124
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 91.0.864.64

! Doctor found issues in 1 category.
```

</details>

Please try the project above on the latest `stable, 2.2.3`

**Comment by Maverick099** [comment]
Okk i refer the your project files and let you know if the issue resolves.

**Comment by Maverick099** [comment]
@TahaTesser  I updated flutter from 2.2.2 to 2.2.3 and also changed index.html to match your index.html it was missing a chunk of lines I think it was implemented in 2.2.3. This resolved the issue. Thank you for your help 😁
